--- ./orig/MPI_Cart_get.c	2019-02-19 13:52:57.000000001 -0500
+++ ./src/MPI_Cart_get.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,94 @@
+/* 
+MPI_Cart_get
+
+   Retrieves Cartesian topology information associated with a communicator
+int MPI_Cart_get(
+  MPI_Comm comm,
+  int maxdims,
+  int *dims,
+  int *periods,
+  int *coords
+);
+
+Parameters
+
+   comm
+          [in] communicator with cartesian structure (handle)
+
+   maxdims
+          [in] length of vectors dims, periods, and coords in the calling
+          program (integer)
+
+   dims
+          [out] number of processes for each cartesian dimension (array of
+          integer)
+
+   periods
+          [out] periodicity (true/false) for each cartesian dimension
+          (array of logical)
+
+   coords
+          [out] coordinates of calling process in cartesian structure
+          (array of integer)
+
+Remarks
+
+   The functions MPI_CARTDIM_GET and MPI_CART_GET return the cartesian
+   topology information that was associated with a communicator by
+   MPI_CART_CREATE.
+
+Thread and Interrupt Safety
+
+   This routine is thread-safe. This means that this routine may be safely
+   used by multiple threads without the need for any user-provided thread
+   locks. However, the routine is not interrupt safe. Typically, this is
+   due to the use of memory allocation routines such as malloc or other
+   non-MPICH runtime routines that are themselves not interrupt-safe.
+
+Notes for Fortran
+
+   All MPI routines in Fortran (except for MPI_WTIME and MPI_WTICK) have
+   an additional argument ierr at the end of the argument list. ierr is an
+   integer and has the same meaning as the return value of the routine in
+   C. In Fortran, MPI routines are subroutines, and are invoked with the
+   call statement.
+
+   All MPI objects (e.g., MPI_Datatype, MPI_Comm) are of type INTEGER in
+   Fortran.
+
+Errors
+
+   All MPI routines (except MPI_Wtime and MPI_Wtick) return an error
+   value; C routines as the value of the function and Fortran routines in
+   the last argument. Before the value is returned, the current MPI error
+   handler is called. By default, this error handler aborts the MPI job.
+   The error handler may be changed with MPI_Comm_set_errhandler (for
+   communicators), MPI_File_set_errhandler (for files), and
+   MPI_Win_set_errhandler (for RMA windows). The MPI-1 routine
+   MPI_Errhandler_set may be used but its use is deprecated. The
+   predefined error handler MPI_ERRORS_RETURN may be used to cause error
+   values to be returned. Note that MPI does not guarentee that an MPI
+   program can continue past an error; however, MPI implementations will
+   attempt to continue whenever possible.
+
+   MPI_SUCCESS
+          No error; MPI routine completed successfully.
+
+   MPI_ERR_TOPOLOGY
+          Invalid topology. Either there is no topology associated with
+          this communicator, or it is not the correct type (e.g., MPI_CART
+          when expecting MPI_GRAPH).
+
+   MPI_ERR_COMM
+          Invalid communicator. A common error is to use a null
+          communicator in a call (not even allowed in MPI_Comm_rank).
+
+   MPI_ERR_ARG
+          Invalid argument. Some argument is invalid and is not identified
+          by a specific error class (e.g., MPI_ERR_RANK).
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
