--- ./orig/MPI_Cart_create.c	2019-02-19 13:52:57.000000001 -0500
+++ ./src/MPI_Cart_create.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,107 @@
+/* 
+MPI_Cart_create
+
+   Makes a new communicator to which topology information has been
+   attached
+int MPI_Cart_create(
+  MPI_Comm comm_old,
+  int ndims,
+  int *dims,
+  int *periods,
+  int reorder,
+  MPI_Comm *comm_cart
+);
+
+Parameters
+
+   comm_old
+          [in] input communicator (handle)
+
+   ndims
+          [in] number of dimensions of cartesian grid (integer)
+
+   dims
+          [in] integer array of size ndims specifying the number of
+          processes in each dimension
+
+   periods
+          [in] logical array of size ndims specifying whether the grid is
+          periodic (true) or not (false) in each dimension
+
+   reorder
+          [in] ranking may be reordered (true) or not (false) (logical)
+
+   comm_cart
+          [out] communicator with new cartesian topology (handle)
+
+Algorithm
+
+   We ignore reorder info currently.
+
+Remarks
+
+   MPI_CART_CREATE returns a handle to a new communicator to which the
+   cartesian topology information is attached. If reorder = false then the
+   rank of each process in the new group is identical to its rank in the
+   old group. Otherwise, the function may reorder the processes (possibly
+   so as to choose a good embedding of the virtual topology onto the
+   physical machine). If the total size of the cartesian grid is smaller
+   than the size of the group of comm, then some processes are returned
+   MPI_COMM_NULL, in analogy to MPI_COMM_SPLIT. The call is erroneous if
+   it specifies a grid that is larger than the group size.
+
+Thread and Interrupt Safety
+
+   This routine is thread-safe. This means that this routine may be safely
+   used by multiple threads without the need for any user-provided thread
+   locks. However, the routine is not interrupt safe. Typically, this is
+   due to the use of memory allocation routines such as malloc or other
+   non-MPICH runtime routines that are themselves not interrupt-safe.
+
+Notes for Fortran
+
+   All MPI routines in Fortran (except for MPI_WTIME and MPI_WTICK) have
+   an additional argument ierr at the end of the argument list. ierr is an
+   integer and has the same meaning as the return value of the routine in
+   C. In Fortran, MPI routines are subroutines, and are invoked with the
+   call statement.
+
+   All MPI objects (e.g., MPI_Datatype, MPI_Comm) are of type INTEGER in
+   Fortran.
+
+Errors
+
+   All MPI routines (except MPI_Wtime and MPI_Wtick) return an error
+   value; C routines as the value of the function and Fortran routines in
+   the last argument. Before the value is returned, the current MPI error
+   handler is called. By default, this error handler aborts the MPI job.
+   The error handler may be changed with MPI_Comm_set_errhandler (for
+   communicators), MPI_File_set_errhandler (for files), and
+   MPI_Win_set_errhandler (for RMA windows). The MPI-1 routine
+   MPI_Errhandler_set may be used but its use is deprecated. The
+   predefined error handler MPI_ERRORS_RETURN may be used to cause error
+   values to be returned. Note that MPI does not guarentee that an MPI
+   program can continue past an error; however, MPI implementations will
+   attempt to continue whenever possible.
+
+   MPI_SUCCESS
+          No error; MPI routine completed successfully.
+
+   MPI_ERR_TOPOLOGY
+          Invalid topology. Either there is no topology associated with
+          this communicator, or it is not the correct type (e.g., MPI_CART
+          when expecting MPI_GRAPH).
+
+   MPI_ERR_DIMS
+          Invalid dimension argument. A dimension argument is null or its
+          length is less than or equal to zero.
+
+   MPI_ERR_ARG
+          Invalid argument. Some argument is invalid and is not identified
+          by a specific error class (e.g., MPI_ERR_RANK).
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
