--- ./orig/MPI_File_seek.c	2019-02-19 13:52:58.000000001 -0500
+++ ./src/MPI_File_seek.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,38 @@
+/* 
+MPI_File_seek
+
+   Updates the individual file pointer
+int MPI_File_seek(
+  MPI_File mpi_fh,
+  MPI_Offset offset,
+  int whence
+);
+
+Parameters
+
+   mpi_fh
+          [in] file handle (handle)
+
+   offset
+          [in] file offset (integer)
+
+   whence
+          [in] update mode (state)
+
+Remarks
+
+   MPI_FILE_SEEK updates the individual file pointer according to whence,
+   which has the following possible values:
+     * MPI_SEEK_SET: the pointer is set to offset
+     * MPI_SEEK_CUR: the pointer is set to the current pointer position
+       plus offset
+     * MPI_SEEK_END: the pointer is set to the end of file plus offset
+
+   The offset can be negative, which allows seeking backwards. It is
+   erroneous to seek to a negative position in the view.
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
