--- ./orig/MPI_File_get_info.c	2019-02-19 13:52:58.000000001 -0500
+++ ./src/MPI_File_get_info.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,37 @@
+/* 
+MPI_File_get_info
+
+   Returns the hints for a file that are actually being used by MPI
+int MPI_File_get_info(
+  MPI_File mpi_fh,
+  MPI_Info *info_used
+);
+
+Parameters
+
+   mpi_fh
+          [in] file handle (handle)
+
+   info_used
+          [out] info object (handle)
+
+Remarks
+
+   MPI_FILE_GET_INFO returns a new info object containing the hints of the
+   file associated with fh. The current setting of all hints actually used
+   by the system related to this open file is returned in info_used. The
+   user is responsible for freeing info_used via MPI_INFO_FREE.
+
+   Advice to users.
+
+   The info object returned in info_used will contain all hints currently
+   active for this file. This set of hints may be greater or smaller than
+   the set of hints passed in to MPI_FILE_OPEN, MPI_FILE_SET_VIEW, and
+   MPI_FILE_SET_INFO, as the system may not recognize some hints set by
+   the user, and may recognize other hints that the user has not set.
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
