--- ./orig/MPI_File_c2f.c	2019-02-19 13:52:58.000000001 -0500
+++ ./src/MPI_File_c2f.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,22 @@
+/* 
+MPI_File_c2f
+
+   Translates a C file handle to a Fortran file handle
+MPI_Fint MPI_File_c2f(
+  MPI_File mpi_fh
+);
+
+Parameters
+
+   mpi_fh
+          [in] C file handle (handle)
+
+Return Value
+
+   Fortran file handle (integer)
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
