--- ./orig/MPI_File_set_info.c	2019-02-19 13:52:58.000000001 -0500
+++ ./src/MPI_File_set_info.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,37 @@
+/* 
+MPI_File_set_info
+
+   Sets new values for the hints associated with a file
+int MPI_File_set_info(
+  MPI_File mpi_fh,
+  MPI_Info info
+);
+
+Parameters
+
+   mpi_fh
+          [in] file handle (handle)
+
+   info
+          [in] info object (handle)
+
+Remarks
+
+   MPI_FILE_SET_INFO sets new values for the hints of the file associated
+   with mpi_fh. MPI_FILE_SET_INFO is a collective routine. The info object
+   may be different on each process, but any info entries that an
+   implementation requires to be the same on all processes must appear
+   with the same value in each process's info object.
+
+   Advice to users.
+
+   Many info items that an implementation can use when it creates or opens
+   a file cannot easily be changed once the file has been created or
+   opened. Thus, an implementation may ignore hints issued in this call
+   that it would have accepted in an open call.
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
