--- ./orig/MPI_File_get_position.c	2019-02-19 13:52:58.000000001 -0500
+++ ./src/MPI_File_get_position.c	2019-01-08 11:09:49.000000001 -0500
@@ -1,3 +1,36 @@
+/* 
+MPI_File_get_position
+
+   Returns the current position of the individual file pointer in etype
+   units relative to the current view
+int MPI_File_get_position(
+  MPI_File mpi_fh,
+  MPI_Offset *offset
+);
+
+Parameters
+
+   mpi_fh
+          [in] file handle (handle)
+
+   offset
+          [out] offset of individual file pointer (nonnegative integer)
+
+Remarks
+
+   MPI_FILE_GET_POSITION returns, in offset, the current position of the
+   individual file pointer in etype units relative to the current view.
+
+   Advice to users.
+
+   The offset can be used in a future call to MPI_FILE_SEEK using whence =
+   MPI_SEEK_SET to return to the current position. To set the displacement
+   to the current file pointer position, first convert offset into an
+   absolute byte position using MPI_FILE_GET_BYTE_OFFSET, then call
+   MPI_FILE_SET_VIEW with the resulting displacement.
+
+*/
+ 
 // Copyright 2009 Deino Software. All rights reserved.
 // Source: http://mpi.deino.net/mpi_functions/index.htm
 
